.github/workflows/terraform-smoke.yml

```diff
@@ -1,45 +1,15 @@
 name: "Terraform Provider for Juju Smoke"
 on:
-  push:
-    branches: [2.9, 3.*]
```

Comment:
We now have the terraform smoke test running in main, where its not expected to succeed.

https://github.com/juju/juju/actions/runs/9962663193/job/27532519680?pr=17753